### PR TITLE
Issue 13483 we need to update the host picker when you archive a site

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java [hunk 1]
```diff
@@ -177,4 +177,18 @@ public Host getCurrentSite(final HttpServletRequest req, final User user) {
 			throw new DotRuntimeException(e);
 		}
 	}
+
+	public void swicthSite(HttpServletRequest req, String hostId) {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java]: This method is misspelled: "swicthSite", should be "switchSite"

File: dotCMS/src/main/java/com/dotcms/util/deserializer/ContentletDeserializer.java [hunk 2]
```diff
@@ -39,6 +41,13 @@ public Object deserialize(JsonElement jsonElement, Type type, JsonDeserializatio
     public JsonElement serialize(Contentlet contentlet, Type type, JsonSerializationContext context) {
         JsonObject jsonObject = (JsonObject) context.serialize( contentlet.getMap() );
         jsonObject.add("lowIndexPriority", new JsonPrimitive( contentlet.isLowIndexPriority()));
+
+        try {
+            jsonObject.add(Contentlet.ARCHIVED_KEY, new JsonPrimitive( contentlet.isArchived() ));
+        } catch (DotDataException|DotSecurityException e) {
```
[the remark this answers is not in the document]
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/util/deserializer/ContentletDeserializer.java]: The exception should be logged, at least at debug level

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java [hunk 3]
```diff
@@ -177,4 +177,18 @@ public Host getCurrentSite(final HttpServletRequest req, final User user) {
...
+		final HttpSession session = req.getSession();
+
+		session.setAttribute(
+				com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID, hostId);
+		session.removeAttribute(WebKeys.CONTENTLET_LAST_SEARCH);
+	}
+
+	public Host swicthToDefaultHost(HttpServletRequest req, User user) throws DotSecurityException, DotDataException {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java]: This method is misspelled: "swicthToDefaultHost", should be "switchToDefaultHost"

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java [hunk 4]
```diff
@@ -197,7 +197,13 @@ public void setInode(String inode) {
      * @return
      */
     public long getLanguageId() {
-    	return (Long)map.get(LANGUAGEID_KEY);
+    	try {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java]: You can do something like ConversionUtils.toLong (map.get(LANGUAGEID_KEY), 0);

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java]: in fact, it already exists

**Review by nollymar — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**